pkg/kubectl/cmd/resource/get.go

```diff
@@ -253,10 +266,11 @@ func (options *GetOptions) Run(f cmdutil.Factory, cmd *cobra.Command, args []str
 		return options.watch(f, cmd, args)
 	}
 
-	printOpts := cmdutil.ExtractCmdPrintOptions(cmd, options.AllNamespaces)
-	printer, err := cmdutil.PrinterForOptions(printOpts)
-	if err != nil {
-		return err
+	// TODO: this block should not exist. We should find a way to use
+	// cmdutil.PrintObject, while knowing if the printer is humanreadable or not.
```

Comment:
Yes, that was exactly the thing I was thinking about this morning. How to know which one to use, although I thought about it in `kubectl get` vs `kubectl describe` command. The flags won't tell you that you should one or the other, but we need a way to do so manually somehow.